amazon-kinesis-client/src/test/java/software/amazon/kinesis/retrieval/fanout/FanOutRecordsPublisherTest.java

```diff
@@ -1112,7 +1112,7 @@ public void testContinuesAfterSequence() {
                         .type(ShardIteratorType.AFTER_SEQUENCE_NUMBER).build())
                 .build();
 
-        verify(kinesisClient).subscribeToShard(eq(nextExpected), nextFlowCaptor.capture());
+        verify(kinesisClient, times(2)).subscribeToShard(any(SubscribeToShardRequest.class), nextFlowCaptor.capture());
```

Comment:
Same as above, we at least need to verify the sequence number we are restarting at, can we use any other data point or object in our test to tell us this information?

Take a look here for how/when to use these in general: https://www.baeldung.com/mockito-argument-matchers 
The custom Argument matcher might be useful for checking on the parameters of the object that we care about.


````
public class SubscribeToShardRequestMatcher implements ArgumentMatcher<SubscribeToShardRequest> {
 
    private SubscribeToShardRequest left;
 
    // constructors
 
    @Override
    public boolean matches(SubscribeToShardRequest right) {
        return left.getShardId().equals(right.getShardId()) &&
          left. getStartingPosition().equals(right.getStartingPosition()) &&
          left. getConsumerARN().equals(right.getConsumerARN());
    }
}
````

And then the verification becomes:

````
verify(kinesisClient).subscribeToShard(argThat(new SubscribeToShardRequestMatcher(nextExpected)), nextFlowCaptor.capture());
````

Reply:
Addressed.